cardinal/ecs/transaction.go

```diff
@@ -54,6 +56,26 @@ func NewTransactionType[In, Out any](
 	name string,
 	opts ...func() func(*TransactionType[In, Out]),
 ) *TransactionType[In, Out] {
+
+	var in In
+	var out Out
+	inType := reflect.TypeOf(in)
+	inKind := inType.Kind()
+	inValid := false
+	if (inKind == reflect.Pointer && inType.Elem().Kind() == reflect.Struct) || inKind == reflect.Struct {
+		inValid = true
+	}
+	outType := reflect.TypeOf(out)
+	outKind := inType.Kind()
+	outValid := false
+	if (outKind == reflect.Pointer && outType.Elem().Kind() == reflect.Struct) || outKind == reflect.Struct {
+		outValid = true
+	}
+
+	if !outValid || !inValid {
+		panic(fmt.Sprintf("Invalid ReadType: %s: The Request and Reply must be both structs", name))
```

Comment:
this should be `TransactionType`, params should be `In` and `Out` as well

Reply:
ah shit yeah, moved too quick.